STACK-UP-3/warriorz-backend
# #171784096 add edit open trip request feature

#### What this PR does?
 -Delivers edit open trip request resources to the project.

#### Description of the task to be completed?

- add updateOpenTrip to tripControllers
- add tests and validations for trips
- add swagger documentation of the endpoints

#### How should this be tested?
 - First, you clone the repo on your machine.
 - Open the folder in your text editor,  *preferably   `visual studio code`*
 - Now open your terminal, *for `visual studio code`    just click on    `terminal`* 
 - Now run `npm install`
 - Then  run `npm run start:dev`,
- First `signup`, verify email, and `sign in`.
- After now create a trip request.
 - Now open `postman` and run endpoint using POST request *`localhost:5000/api/v1/trips/:trip_id`* route; 
OR
 - You can test it on the swagger Documentation(*in your browser*) using the `http://localhost:5000/api-docs/` route.

#### What are the relevant pivotal tracker stories?
  [#171784096](https://www.pivotaltracker.com/story/show/171784096)

#### Screenshots?
Creating a trip:
1. PostMan:
![image](https://user-images.githubusercontent.com/52471122/79606639-3225cb00-80f2-11ea-9855-3c3a6f147214.png)

2. DataBase:
![image](https://user-images.githubusercontent.com/52471122/79606935-aa8c8c00-80f2-11ea-95f2-fc81bfc5bfc9.png)


Updating the Trip:
1. Postman
![image](https://user-images.githubusercontent.com/52471122/79607030-db6cc100-80f2-11ea-9818-eded91a0bc17.png)

2.Database
![Untitled](https://user-images.githubusercontent.com/52471122/79607754-0e638480-80f4-11ea-8e1f-7dc010b58462.png)
